CONTRIBUTING.rst

```diff
@@ -841,29 +841,106 @@ Extras
 ------
 
 There are a number of extras that can be specified when installing Airflow. Those
-extras can be specified after the usual pip install - for example
-``pip install -e .[ssh]``. For development purpose there is a ``devel`` extra that
-installs all development dependencies. There is also ``devel_ci`` that installs
-all dependencies needed in the CI environment.
+extras can be specified after the usual pip install - for example ``pip install -e.[editable_ssh]`` for editable
```

Comment:
In the latest version of the change - we have neitherseparated `devel-` nor `editable-` extras. I managed to simplify it a LOT (!!!!) and now simply `.[amazon]` will install amazon dependencies + amazon devel dependencies together in editable mode, and `apache-airflow[amazon]` will install `apache-airflow-providers-amazon` provider only.

This is actually the best of all worlds together, because we do not have to learn about any new extras and developers will simply use:

```bash
pip install --editable .[amazon]
```

and users:

```bash
pip install apache-airflow[amazon] 
```

And it will **just work**(TM) :D

So basically this change means -- no changes to any habits of yourrs. Just do what you did so far, but it will as you expected it to work without any tricks (INSTALL_PROVIDERS_FROM_SOURCES) needed. In fact it will all be 100% compatible, simply INSTALL_PROVIDERS_FROM_SOURCES will be completely ignored, and the decision whether to install `dependencies` (including devel ones per-providers) or `provider` will be purely based on whether you are installing airflow with `--editable` flag or not.